### DISCUSSION TITLE
Lesson 1 - Injected Web3

### DISCUSSION THREAD

**Discussion by redshoes** [smartcontractkit/full-blockchain-solidity-course-py#1179] [Q&A] [answered]
I have the Metamask browser extension installed and connected to my wallet on the Rinkeby Test Network, and have compiled and deployed my smart contract using Remix as per Lession 1.  But when I select "Injected Web 3" as the environment, I get the error "Currently you have no contract instances to interact with." and the Javascript VM is reselected. Metamask doesn't pop up. I have tried in Chrome and Firefox running on Ubuntu 21.10.

**Comment by charlesAuthentik** [comment, marked as answer]
Sometimes you have to manually click the metamask icon to have it "pop up"

**Reply by redshoes** [reply]
Got it, thanks!

**Reply by cromewar** [reply]
Nice catch!!

**Reply by sudolink** [reply]
merely opening metamask didn't help in my case. I had to open the wallet, click the three dots, go to **Connected sites**, then manually connect to current site. Then I had to restart the browser (was on Brave), and that did it, remix then recognized the wallet and pulled the wallet data.
